internal/httpapi/server.go

```diff
@@ -90,6 +101,49 @@ func (c *ServerController) HandleRegister(ctx *gin.Context) {
 	ctx.JSON(http.StatusCreated, gin.H{"result": "registered"})
 }
 
+// HandleRegister is a gin HTTP handler that allows servers to register
+// themselves in the repository. This call will also dial back to the port
+// being registered and confirm that the port is accessible.
+func (c *ServerController) HandleRegister(ctx *gin.Context) {
+	serverAddr, err := srvrepo.ParseServerAddress(ctx.Param("server_id"))
+	if err != nil {
+		// 404, since the ID is a URL param
+		ctx.JSON(http.StatusNotFound, gin.H{"result": "invalid server ID"})
```

Comment:
Malformed server ID is one problem and not found is another. If it's malformed, is there a better way of passing `err` back to the client in this 404?

Reply:
Now sending BadRequest 400